tools/pycbc_test_suite.sh

```diff
@@ -44,10 +44,28 @@ if [ "$PYCBC_TEST_TYPE" = "help" ] || [ -z ${PYCBC_TEST_TYPE+x} ]; then
         else
             echo -e "    Pass."
         fi
+	if [[ `echo $prog | egrep '(pycbc_copy_output_map|pycbc_submit_dax|pycbc_stageout_failed_workflow)'` ]] ; then
+	    continue
+	fi
+	echo -e ">> [`date`] running $prog --version"
+	$prog --version &> $LOG_FILE
+	if test $? -ne 0 ; then
+            RESULT=1
+            echo -e "    FAILED!"
+            echo -e "---------------------------------------------------------"
+            cat $LOG_FILE
+            echo -e "---------------------------------------------------------"
+        else
+            echo -e "    Pass."
+        fi
     done
-    # also check that --version works for one executable
-    echo -e ">> [`date`] running pycbc_inspiral --version"
+    # also check that --version with increased modifiers works for one executable
+    echo -e ">> [`date`] running pycbc_inspiral --version with modifiers"
     pycbc_inspiral --version &> $LOG_FILE
+    pycbc_inspiral --version 0 &> $LOG_FILE
+    pycbc_inspiral --version 1 &> $LOG_FILE
+    pycbc_inspiral --version 2 &> $LOG_FILE
+    pycbc_inspiral --version 3 &> $LOG_FILE
```

Comment:
This will miss failure modes that are triggered by all invocations except for the last one, since the last one will succeed and set `$?` to zero regardless of the previous ones. I suggest doing a for loop with a similar structure to the loop just above.

Reply:
I.e.
```bash
for modifier in "" 0 1 2 3`
do
    pycbc_inspiral --version ${modifier} &> $LOG_FILE
    if test $? -ne 0
        …
    fi
done
```

Reply:
Thats a really good point. 

I've also move all of the `if test $? -ne 0 ; then` statement into its own function, as it is reused _a lot_ in this script

Reply:
Does moving to a function really work though? Does bash let you change a global variable like that from inside a function?

Reply:
yup, I checked as I had the same worry - so long as the `local` keyword isnt set within the function, it should work

Reply:
Nope, that does not work:
```
(base) tito@newman:~ $ cat foo.sh 
ANTANI=0

echo $ANTANI

function set_antani {
    ANTANI=1
}

echo $ANTANI
(base) tito@newman:~ $ bash foo.sh 
0
0
```